auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.java

```diff
@@ -318,6 +321,45 @@ public AuthenticationRequest loginWithOAuthAccessToken(@NonNull String token, @N
                 .addAuthenticationParameters(parameters);
     }
 
+    /**
+     * Log in a user using a token obtained from a Native Social Identity Provider, such as Facebook, using <a href="https://auth0.com/docs/api/authentication#token-exchange-for-native-social">'\oauth\token' endpoint</a>
+     * The default scope used is 'openid'.
+     * Example usage:
+     * <pre>
+     * {@code
+     * client.loginWithNativeSocialToken("{subject token}", "{subject token type}")
+     *      .start(new BaseCallback<Credentials>() {
+     *          {@literal}Override
+     *          public void onSuccess(Credentials payload) { }
+     *
+     *          {@literal}Override
+     *          public void onFailure(AuthenticationException error) { }
+     *      });
+     * }
+     * </pre>
+     *
+     * @param token     the subject token, typically obtained through the Identity Provider's SDK
+     * @param tokenType the subject token type that is associated with this Identity Provider. e.g. 'http://auth0.com/oauth/token-type/facebook-session-access-token'
+     * @return a request to configure and start that will yield {@link Credentials}
+     */
+    @SuppressWarnings("WeakerAccess")
+    public AuthenticationRequest loginWithNativeSocialToken(@NonNull String token, @NonNull String tokenType) {
+        HttpUrl url = HttpUrl.parse(auth0.getDomainUrl()).newBuilder()
+                .addPathSegment(OAUTH_PATH)
+                .addPathSegment(TOKEN_PATH)
+                .build();
+
+        Map<String, Object> parameters = ParameterBuilder.newAuthenticationBuilder()
+                .setGrantType(GRANT_TYPE_TOKEN_EXCHANGE)
+                .setClientId(getClientId())
+                .set(SUBJECT_TOKEN_KEY, token)
+                .set(SUBJECT_TOKEN_TYPE_KEY, tokenType)
+                .asDictionary();
+
+        return factory.authenticationPOST(url, client, gson)
+                .addAuthenticationParameters(parameters);
```

Comment:
Just as this is adding parameters, the returned request instance can also be added more parameters. This covers support for audience, scope, user profiles, etc.